client.go

```diff
@@ -0,0 +1,249 @@
+package libhoney
+
+import (
+	"errors"
+
+	"github.com/honeycombio/libhoney-go/transmission"
+)
+
+// Client represents an object that can create new builders and events and send
+// them somewhere. It maintains its own sending queue for events, distinct from
+// both the package-level libhoney queue and any other client. Clients should be
+// created with NewClient(config). A manually created Client{} will function as
+// a nil output and drop everything handed to it (so can be safely used in dev
+// and tests). For more complete testing you can create a Client with a
+// MockOutput transmission then inspect the events it would have sent.
+type Client struct {
+	transmission transmission.Sender
+	logger       Logger
+	builder      *Builder
```

Comment:
Is a `Builder` something that's valid even when empty? If so, you could just have type `Builder` here and then you'd never confront the `builder == nil` problem.

Reply:
I put some `sync.Once`es in the `ensure`s. An empty builder isn't really a thing I want to start mucking with; they should be created by cloning an existing builder or using `NewBuilder()`.